### ISSUE TITLE
附录B中交通元素识别，代码如何实现

### ISSUE THREAD

**Issue by yu-yanjia** [QwenLM/Qwen-Drive-1.0#5]
请问附录中B.2 Traffic-Signal Detection 和其他对交通要素的识别， 是通过python scripts/run_vqa.py --model Qwen-Drive-1.0-4B --image front.jpg --question ‘论文中prompt’ 实现的吗，如果不是，请教下附录中对交通元素识别的代码方法，不胜感激。

**Comment by LMD0311** [comment]
Yes. The traffic-signal and other traffic-element detections were performed using the prompts provided in the paper, following the same invocation method as other VLMs.

**Comment by yu-yanjia** [comment]
非常感谢您的回答，我这样测试了交通灯，图片用的是data/demo中的 question输入的是论文中的prompt，（我也尝试将question置空，论文中的promot赋值给processor.encode_vqa中的system），检测效果均不好，我是不是哪里做错了或者漏做了什么。

**Comment by LMD0311** [comment]
I just tried `data/demo/frames/scene0_0.jpg` and `data/demo/frames/scene0_11.jpg` with this prompt：
```
You are a visual annotation assistant for autonomous-driving intersection scenes. Given an image, detect and output all traffic-signal-related objects, focusing on the signal type (label), orientation (direction), and relevance to the current intersection (is_related).

For every traffic-signal-related object in the image, output a dictionary containing:

- bbox_2d: the 2D bounding box in the format [x1, y1, x2, y2] using integer pixel coordinates. The top-left corner of the image is (0,0), with x1 < x2 and y1 < y2. Make the box as tight as possible around the visible signal body.
- label: the signal type, one of "car" (motor-vehicle or non-motor-vehicle signal), "pedestrian" (pedestrian signal), "black" (black, unlit signal), "digit" (standalone countdown display), "led" (text LED sign), and "other" (another traffic signal or signal-control device).
- direction: the signal orientation, one of "front" (facing the ego vehicle), "back" (facing away from the ego vehicle), and "side" (clearly facing sideways).
- is_related: whether the signal is relevant to the current intersection decision, one of "related" (controls traffic through the current intersection) and "unrelated" (a signal for a distant unrelated intersection, opposing traffic, or a side road).

Output exactly one JSON array of dictionaries with no extra explanatory text, wrapped in ```json ... ```. If there are no traffic-signal-related objects, output an empty array. Output each object only once, and keep bounding-box coordinates as integers.
```

The outputs are:
```json
[
        {"bbox_2d": [419, 499, 461, 551], "label": "car", "direction": "front", "is_related": "related"},
        {"bbox_2d": [659, 492, 690, 541], "label": "car", "direction": "front", "is_related": "related"}
]
```
and
```json
[
        {"bbox_2d": [186, 287, 223, 366], "label": "car", "direction": "front", "is_related": "related"},
        {"bbox_2d": [327, 242, 362, 314], "label": "car", "direction": "front", "is_related": "related"},
        {"bbox_2d": [738, 507, 759, 541], "label": "pedestrian", "direction": "front", "is_related": "related"}
]
```

This task used only a small amount of data, and we actually trained on different prompts. Additionally, we did not train on the WOD dataset (or any other public dataset) for this task. However, **the prediction results still look quite good**. I suspect you may not have handled the coordinate system correctly (we used normalized coordinates ranging from 0 to 1000). If you need better performance in specific scenarios, further fine-tuning may be required.

<img width="972" height="1079" alt="Image" src="https://github.com/user-attachments/assets/b6d39d05-4aec-460e-b3fb-d3629bc96311" />
<img width="972" height="1079" alt="Image" src="https://github.com/user-attachments/assets/63877e8d-72dd-4755-9e3d-0576fbfc7a88" />

**Comment by yu-yanjia** [comment]
非常感谢您的回答，我确实没有norm ，norm后是正确的， 非常棒的工作
